### PR TITLE
Add retries gem and use it on SimpleDB 503 errors.

### PR DESCRIPTION
**Pull request by philiplrb**

### CODE REVIEW

File: lib/heirloom/aws/simpledb.rb [hunk 1]
```diff
@@ -80,17 +111,32 @@ def select(query, opts = {})
...
+        end
       end
 
       def item_count(domain, item)
-        query = "SELECT count(*) FROM `#{domain}` WHERE itemName() = '#{item}'"
-        @sdb.select(query).body['Items']['Domain']['Count'].first.to_i
+        with_retries(:max_retries => 3,
+                     :rescue => Excon::Errors::ServiceUnavailable,
+                     :base_sleep_seconds => 10,
+                     :max_sleep_seconds => 60) do
```
Comment (brettweavnet-intuit) [inline lib/heirloom/aws/simpledb.rb]: consider options moving to it's own private method to dry out code:

```ruby
def retries_options
 {:max_retries => 3,
  :rescue => Excon::Errors::ServiceUnavailable,
  :base_sleep_seconds => 10,
  :max_sleep_seconds => 60}
end
```

**Comment by brettweavnet-intuit** [comment]
@philiplrb one comment, overall I'm good with it, though at some point we should consider switching to AWS CLI rather than continuing investment in heirloom.

**Comment by bradly** [comment]
:+1: Looks good to me.

**Comment by philiplrb** [comment]
@brettweavnet-intuit: Agree, but this seems like the shortest path to get LC building again.

**Pull request merged**